lib/index.js

```diff
@@ -224,13 +226,25 @@ Socket.prototype._register = function(chan){
  */
 
 Socket.prototype.bind = function (addr) {
-    return this._protect(nn.Bind(this.binding, addr));
+  return this.bound[addr] = this._protect(nn.Bind( this.binding, addr ));
 }
 
 Socket.prototype.connect = function (addr) {
-    return this._protect(nn.Connect(this.binding, addr));
+  return this.connected[addr] = this._protect(nn.Connect( this.binding, addr ));
 }
 
+Socket.prototype.shutdown = function (addr) {
+  var eid;
+  if (this.bound.hasOwnProperty(addr)) {
+    eid = this.bound[addr];
+    delete this.bound[addr];
+  } else if (this.connected.hasOwnProperty(addr)) {
+    eid = this.connected[addr];
+    delete this.connected[addr];
+  }
+  return this._protect(nn.Shutdown(this.binding, eid));
```

Comment:
I think this should go in both branches, otherwise, you're always invoking nn.Shutdown even when ei is undefined.